src/library/components/ProductCard.tsx

```diff
@@ -0,0 +1,167 @@
+  image?: ReactElement< ImageProps >;
+  gradientBrand?: boolean;
+  dropShadow?: boolean;
+}
+
+const ProductCard = (props: Props): JSX.Element => {
+  const { tag, texts, headingLevel, linkTo, variant, assertiveTitle = false, image, gradientBrand, dropShadow} = props;
+
+  const [clonedImage, setClonedImage] = useState<ReactElement>();
+  const [clonedTag, setClonedTag] = useState<ReactElement>();
+
+  const productNameId = useId();
+
+  let cardStyles = {
+    classes: "",
+    titleLinkClasses: "",
+    linkClasses: ""
+  }
+
+  const defaultStyle = {
+    classes: "bsds-card",
+    titleLinkClasses: "bsds-link",
+    linkClasses: "bsds-product-card-ns-title"
+  }
+
+  const ghostStyle = {
+    classes: "bsds-card-ghost",
+    titleLinkClasses: "bsds-link-ghost",
+    linkClasses: "bsds-product-card-ns-title-ghost"
+  }
+
+  const borderLightStyle = {
+    classes: "bsds-card-border-light",
+    titleLinkClasses: "bsds-link-subtle",
+    linkClasses: "bsds-product-card-ns-title"
+  }
+
+  const borderGhostStyle = {
+    classes: "bsds-card-border-ghost",
+    titleLinkClasses: "bsds-link-ghost",
+    linkClasses: "bsds-product-card-ns-title-ghost"
+  }
+
+  switch(variant) {
+    case "ghost":
+      cardStyles = {...ghostStyle};
+      break;
+    case "border-light":
+      cardStyles = {...borderLightStyle};
+      break;
+    case "border-ghost":
+      cardStyles = {...borderGhostStyle};
+      break;
+
+    default:
+      cardStyles = {...defaultStyle};
+      break;
+  }
+
+  let decorativeState = "";
+  if (gradientBrand && variant?.includes("ghost")) {
+    decorativeState = "bsds-card-gradient";
+    cardStyles = {...borderGhostStyle};
+  } else if (gradientBrand) {
+    decorativeState = "bsds-card-gradient";
+    cardStyles = {...borderLightStyle};
+  } else if (dropShadow && variant?.includes("ghost")) {
+    decorativeState = "bsds-card-shadow";
+    cardStyles = {...borderGhostStyle};
+  } else if (dropShadow) {
+    decorativeState = "bsds-card-shadow";
+    cardStyles = {...borderLightStyle};
+  }
+
+  useEffect(() => {
+    if(tag) {
+      setClonedTag(cloneElement(tag as ReactElement, {
+        isGhost: (variant === "ghost" || variant === "border-ghost"),
+     }));
+    }
+  },[tag, variant])
+
+  useEffect(() => {
+    if(image) {
+      setClonedImage(cloneElement(image as ReactElement, {
+        ratio: (image.props.ratio ? image.props.ratio : "1:1"),
+        isDecorative: false,
+        isGhost: (variant === "ghost" || variant === "border-ghost"),
+     }));
+    }
+  },[image, linkTo, variant])
+
+  useEffect(() => {
+    //This is to expand the link hitbox for 50% split image
+    if(clonedImage?.props.ratio === "50:50") {
+      const productCardLinks = document.querySelectorAll(".bsds-product-card-even-split");
+      Array.prototype.forEach.call(productCardLinks, productCardLink => {
+        let link = productCardLink.querySelector("a");
+        let cardImg = productCardLink.querySelector("img");
```

Comment:
I'm wondering if we could use refs here. And add the onclick to the image when its cloned.